server/server.go

```diff
@@ -387,11 +387,37 @@ func New(ctx context.Context, config *Config) (*Server, error) {
 
 	hostIP := net.ParseIP(config.HostIP)
 	if hostIP == nil {
-		hostIP, err = knet.ChooseBindAddress(nil)
+		// First, check if we can find an IP address unambiguously
+		allAddrs, err := net.InterfaceAddrs()
 		if err != nil {
 			return nil, err
 		}
+
+		// we are hoping for exactly 1 possible IP
+		numPossibleIPs := 0
+		// adapted from: https://stackoverflow.com/a/31551220
+		for _, addr := range allAddrs {
+			if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+				if ipnet.IP.To4() != nil {
+					numPossibleIPs++
+					hostIP = ipnet.IP
+					if numPossibleIPs > 1 {
+						break
+					}
+				}
+			}
+		}
+
+		// We either found no hostIP, or too many. Attempt to find a
+		// primary IP from /proc/net/route deterministically
+		if numPossibleIPs != 1 {
+			hostIP, err = knet.ChooseBindAddress(nil)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
lets fall through to one more case.  if `ChooseBindAddress` fails, but `numPossibleIPs != 0`, lets just choice the first IPv4 address.  @mrunalp what do you think?

Reply:
per internal conversations, we've decided this could lead to restarts of CRI-O having addresses changed from under it. marking as resolved